cmd/migratecheckpoint/log_library.go

```diff
@@ -29,8 +29,9 @@ type Fingerprint struct {
 }
 
 type Reader struct {
-	Fingerprint *Fingerprint
-	Offset      int64
+	Fingerprint    *Fingerprint
+	Offset         int64
+	FileAttributes map[string]any
```

Comment:
what FileAttributes doing?

Reply:
We're trying to convert fluentd's format to the following struct as per OTeL's format. We get `Offset` and `Fingerprint` by reading the fluentd's pos_file. But the field `FileAttributes` is left nil. 
This causes a panic when the filelogreceiver tries to access this field while decoding it (causes nil reference). Setting it to an empty map resolves this.

<img width="462" alt="Screenshot 2023-11-04 at 1 28 06 AM" src="https://github.com/signalfx/splunk-otel-collector/assets/121151420/1358c33d-3fda-47cb-8e46-008d22cae415">

Reply:
Makes sense, thanks!